### PR TITLE
15/apps query client

### LINKED ISSUE #15: Adapt File drop for New Graasp backend

### PR DESCRIPTION
**Pull request by pyphilia**
Another big one...

close #15

### CODE REVIEW

File: src/utils/uppy.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
+/* eslint-disable no-unused-vars */
```
Comment (juancarlosfarah) [inline src/utils/uppy.js:1]: What is this disabling? Maybe use it per line if needed.

File: cypress/fixtures/members.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    type: 'individual',
+  },
+  BOB: {
+    id: '1f0a2774-a965-4b97-afb4-bccc3796e060',
+    name: 'bob',
+    type: 'individual',
+  },
+};
+
+export const CURRENT_MEMBER = MEMBERS.ANNA;
```
Comment (juancarlosfarah) [inline cypress/fixtures/members.js:14]: At some point we could factor out these fixtures to some common repo.

File: cypress/support/index.js [hunk 3]
```diff
@@ -13,8 +13,14 @@
...
 
-// Alternatively you can use CommonJS syntax:
-// require('./commands')
+Cypress.on('uncaught:exception', (err, runnable, promise) => {
+  // returning false here prevents Cypress from failing the test
+  // this is necessary to accept wanted error from mirage to fail the test
+  if (promise) {
+    return false;
+  }
+});
```
Comment (juancarlosfarah) [inline cypress/support/index.js:26]: Could this potentially block us from finding real errors?

File: src/components/App.js [hunk 4]
```diff
@@ -1,41 +1,35 @@
+/* eslint-disable no-unused-vars */
```
Comment (juancarlosfarah) [inline src/components/App.js:1]: Why do we need this here?

File: src/components/App.js [hunk 5]
```diff
@@ -1,41 +1,35 @@
...
 
-  if (token != null) {
-    switch (mode) {
-      // show teacher view when in producer (educator) mode
-      case 'teacher':
-      case 'producer':
-      case 'educator':
-      case 'admin':
+  const renderContent = () => {
+    switch (context?.get('permission', DEFAULT_PERMISSION)) {
```
Comment (juancarlosfarah) [inline src/components/App.js:14]: Technically, we should first have a switch for the context (builder / player), right?

File: src/components/context/ContextContext.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      i18n.changeLanguage(lang);
+    }
+  }, [context]);
+
+  if (isLoading) {
+    return <Loader />;
+  }
+
+  if (isError) {
+    toast.error('An error occured while fetching the context.');
```
Comment (juancarlosfarah) [inline src/components/context/ContextContext.js:27]: `t()`?

File: src/components/context/TokenContext.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  const { data, isLoading, isError } = hooks.useAuthToken({
+    origin: window.location.origin,
+  });
+
+  if (isLoading) {
+    return <Loader />;
+  }
+
+  if (isError) {
+    toast.error('An error occured while requesting the token.');
```
Comment (juancarlosfarah) [inline src/components/context/TokenContext.js:19]: `t()`

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
👏 

😅  Just some minor comments and questions!

File: cypress/support/index.js (hunk 3 above)
Comment (pyphilia) [reply cypress/support/index.js:26]: Potentially yes. I will add some console so the user can more easily debug.

**Review by juancarlosfarah — APPROVED** [review]
👍

**Pull request merged**